com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/configuration/ActivitiConfiguration.java

```diff
@@ -93,4 +100,19 @@ protected void scale(DefaultJobExecutor jobExecutor) {
         jobExecutor.setMaxPoolSize(JOB_EXECUTOR_MAX_POOL_SIZE);
     }
 
+    @Inject
+    @Bean
+    public String jobExecutorId(ApplicationConfiguration applicationConfiguration) {
+        String applicationGuid = applicationConfiguration.getApplicationGuid();
+        Integer applicationInstanceIndex = applicationConfiguration.getApplicationInstanceIndex();
+        if (applicationGuid == null || applicationInstanceIndex == null) {
+            return randomIdGenerator.get();
+        }
+        return generateJobExecutorId(applicationGuid, applicationInstanceIndex);
+    }
+
+    private String generateJobExecutorId(String applicationGuid, int applicationInstanceIndex) {
```

Comment:
isnt this building id not generating?

Reply:
Mmmm, maybe? I don't really know. :D Part of it is generated, part of it is built. Since 2/3rds of the ID are "built" I'll rename it to buildJobExecutorId.